src/assets/scripts/client/constants/themes/default/modules/dataBlock.js

```diff
@@ -101,6 +101,30 @@ export const DATA_BLOCK_THEME = {
      */
     HAS_FILL: false,
 
+    /**
+     * One half of the height of a data block
+     *
+     * @memberof DATA_BLOCK_THEME
+     * @property HALF_HEIGHT
+     */
+    HALF_HEIGHT: 16,
+
+    /**
+     * One half of the width of a data block
+     *
+     * @memberof DATA_BLOCK_THEME
+     * @property HALF_HEIGHT
+     */
+    HALF_WIDTH: 30,
+
+    /**
+     * Height of a data block
+     *
+     * @memberof DATA_BLOCK_THEME
+     * @property HEIGHT
+     */
+    HEIGHT: 32,
```

Comment:
The goal here was to avoid calculations.  Remember, these methods live inside the loop which means they are going to fire up to 60 times *per second*.  If we are calculating that every frame that gets expensive. 

That said, this doesn't look good.  I'll look into it.

Reply:
done.  

In my opinion, we should do away with this legacy data block entirely some day.  it's a lot of extra logic for, what I'm guessing is, not a lot of use.  We'll see what comes of #1197 and go from there.